Fixtures are called directly error

The update to pytest 4.0 disallows (for no reason whatsoever) a feature we use in the tests, namely calling a fixture inside another fixture, as in

```
======================================================= ERRORS =======================================================
_______________________________________ ERROR at setup of test_chi2_arithmetic _______________________________________

weighted_data = (PDF(name='NNPDF31_nnlo_as_0118'), [ExperimentSpec(name='NMC Experiment', datasets=(DataSetSpec(name='NMC', commondata...Spec(id=162, path=PosixPath('/home/zah/anaconda3/share/NNPDF/data/theory_162')), cuts=None, op='NULL', weight=100),))])

    @pytest.fixture(scope='module')
    def convolution_results_with_weights(weighted_data):
>       return convolution_results(weighted_data)
E       _pytest.warning_types.RemovedInPytest4Warning: Fixture "convolution_results" called directly. Fixtures are not meant to be called directly, are created automatically when test functions request them as parameters. See https://docs.pytest.org/en/latest/fixture.html for more information.
```

We should probably define normal functions that do the same, just to increase the noise. In the meantime, I'll specify pytest < 4 in the conda recipe.

--- a/conda-recipe/meta.yaml
+++ b/conda-recipe/meta.yaml
@@ -42,7 +42,7 @@ requirements:
 test:
     requires:
         - hypothesis
-        - pytest <4 # See https://github.com/NNPDF/nnpdf/issues/335
+        - pytest
         - coverage
         #Build dependencies for catch tests
         - {{ compiler("cxx") }}

--- a/validphys2/src/validphys/tests/conftest.py
+++ b/validphys2/src/validphys/tests/conftest.py
@@ -34,10 +34,13 @@ def data():
     pdf = l.check_pdf("NNPDF31_nnlo_as_0118")
     return pdf, exps
 
-@pytest.fixture(scope='module')
-def convolution_results(data):
+def convolution_results_implement(data):
     pdf, exps = data
     return [results.experiment_results(exp, pdf, pdf) for exp in exps]
+
+@pytest.fixture(scope='module')
+def convolution_results(data):
+    return convolution_results_implement(data)
 
 @pytest.fixture
 def dataset_t0_convolution_results(data):
@@ -65,9 +68,12 @@ def dataset_convolution_results(single_exp_data):
 def dataset_chi2data(dataset_convolution_results):
     return [results.abs_chi2_data(r) for r in dataset_convolution_results]
 
+def chi2data_implement(convolution_results):
+    return [results.abs_chi2_data_experiment(r) for r in convolution_results]
+
 @pytest.fixture(scope='module')
 def chi2data(convolution_results):
-    return [results.abs_chi2_data_experiment(r) for r in convolution_results]
+    return chi2data_implement(convolution_results)
 
 @pytest.fixture(scope='module')
 def weighted_data():
@@ -82,8 +88,8 @@ def weighted_data():
 
 @pytest.fixture(scope='module')
 def convolution_results_with_weights(weighted_data):
-    return convolution_results(weighted_data)
+    return convolution_results_implement(weighted_data)
 
 @pytest.fixture(scope='module')
 def weighted_chi2data(convolution_results_with_weights):
-    return chi2data(convolution_results_with_weights)
+    return chi2data_implement(convolution_results_with_weights)

